cmd/helper/check_provider_update.go

```diff
@@ -0,0 +1,115 @@
+package helper
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/blang/semver"
+	"github.com/loft-sh/devpod/cmd/flags"
+	"github.com/loft-sh/devpod/pkg/config"
+	"github.com/loft-sh/devpod/pkg/log"
+	"github.com/loft-sh/devpod/pkg/provider"
+	"github.com/loft-sh/devpod/pkg/workspace"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+var (
+	errProviderNotFound = errors.New("provider not found")
+)
+
+type CheckProviderUpdateCmd struct {
+	*flags.GlobalFlags
+	log log.Logger
+}
+
+type providerVersionCheck struct {
+	UpdateAvailable bool   `json:"updateAvailable"`
+	LatestVersion   string `json:"latestVersion,omitempty"`
+}
+
+// NewCheckProviderUpdateCmd creates a new command
+func NewCheckProviderUpdateCmd(flags *flags.GlobalFlags) *cobra.Command {
+	cmd := &CheckProviderUpdateCmd{
+		GlobalFlags: flags,
+		log:         log.Default,
+	}
+	shellCmd := &cobra.Command{
+		Use:   "check-provider-update",
+		Short: "Check if a provider update is available",
+		RunE: func(_ *cobra.Command, args []string) error {
+			devPodConfig, err := config.LoadConfig(cmd.Context, cmd.Provider)
+			if err != nil {
+				return err
+			}
+			return cmd.Run(context.Background(), devPodConfig, args)
+		},
+	}
+
+	return shellCmd
+}
+
+func (cmd *CheckProviderUpdateCmd) Run(ctx context.Context, devPodConfig *config.Config, args []string) error {
+	if len(args) != 1 {
+		return fmt.Errorf("provider is missing")
```

Comment:
This could also be rewritten to in line 22

```go
errProviderMissing = errors.New("provider is missing")
```

and then

```suggestion
		return errProviderMissing
```